src/API/Management/Users.php

```diff
@@ -1,88 +1,265 @@
+     */
+    public function getAll($params = [], $fields = null, $include_fields = null, $page = null, $per_page = null)
+    {
+        $params = is_array($params) ? $params : [];
 
-        $client = $this->apiClient->get()
-            ->users();
+        // Fields to include/exclude.
+        if (!isset($params['fields']) && null !== $fields) {
+            $params['fields'] = $fields;
+        }
+        if (isset($params['fields'])) {
+            if (is_array($params['fields'])) {
+                $params['fields'] = implode(',', $params['fields']);
+            }
+            if (null !== $include_fields) {
+                $params['include_fields'] = (bool) $include_fields;
+            }
+        }
 
-        foreach ($params as $param => $value) {
-            $client->withParam($param, $value);
+        // Keep existing pagination params if passed (backwards-compat), override with non-null function param if not.
+        if (!isset($params['page']) && null !== $page) {
+            $params['page'] = abs(intval($page));
+        }
+        if (!isset($params['per_page']) && null !== $per_page) {
+            $params['per_page'] = abs(intval($per_page));
         }
 
-        return $client->call();
+        return $this->apiClient->method('get')
+            ->addPath('users')
+            ->withDictParams($params)
+            ->call();
     }
 
-    public function delete($user_id) {
+    /**
+     * Wrapper for self::search().
+     *
+     * TODO: Deprecate, replaced with getAll above.
+     *
+     * @param array $params - Search parameters to send.
+     *
+     * @return mixed|string
+     *
+     * @throws \Exception
+     *
+     * @see self::getAll()
+     */
+    public function search($params = [])
+    {
+        return $this->getAll($params);
+    }
 
-        return $this->apiClient->delete()
-            ->users($user_id)
+    /**
+     * Delete a User by ID.
+     * Requires scope: delete:users.
+     *
+     * @param string $user_id - User ID to delete.
+     *
+     * @return mixed|string
+     *
+     * @throws \Exception
+     *
+     * @link https://auth0.com/docs/api/management/v2#!/Users/delete_users_by_id
+     */
+    public function delete($user_id)
+    {
+        return $this->apiClient->method('delete')
+            ->addPath('users', $user_id)
             ->call();
     }
 
-    public function linkAccount($user_id, $post_identities_body) {
-
-        return $this->apiClient->post()
-            ->users($user_id)
-            ->identities()
-            ->withHeader(new ContentType('application/json'))
-            ->withBody(json_encode($post_identities_body))
+    /**
+     * Link one user identity to another.
+     * Requires scope: update:users.
+     *
+     * @param string $user_id - User ID of the primary identity
+     * @param array $data - Secondary identity to link; either link_with JWT or provider, connection_id, and user_id.
```

Comment:
What about adding something like "see ways of calling this endpoint in the @link below", like you did for users search v3 above. Or at least add a bulleted list below the data param to differentiate both payloads. (I still find it confusing) 

i.e.
```
- Sending the JWT as the link_with parameter
- Sending the provider, connection_id and user_id values.
```